File: pset6/mario/more/mario.py
#!/usr/bin/env python3
from cs50 import get_int

# function to get valid Height value for mario game


def get_valid_integer():
    height = get_int("Height: ")
    while height < 0 or height > 23:
        height = get_int("Height: ")
    return height

# function for printing blocks


def printBlocks(numberOfBlocks):
    for i in range(numberOfBlocks):
        print("#", end="")


#  get valid height value
height = get_valid_integer()

# two inner loops for pyramid drawing
for row in range(height):
    # counter for blocks in left side
    numberOfBlocks = 0
    for column in range(height):
        # condition to form right pyramid
        if column < height - 1 - row:
            print(" ", end="")
        else:
            print("#", end="")
            numberOfBlocks += 1
    # hole in pyramid center
    print("  ", end="")
    # create right side of pyramid
    printBlocks(numberOfBlocks)
    print()

File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int get_valid_integer();
void printBlocks(int numberOfBlocks);

int main(void)
{
    // get valid height value
    int height = get_valid_integer();

    //two inner loops for pyramid drawing
    for (int row = 0; row < height; row++)
    {
        // counter for blocks in left side
        int numberOfBlocks = 0;
        for (int column = 0; column < height; column++)
        {
            // condition to form right pyramid
            if (column < height - row - 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
                numberOfBlocks++;
            }

        }

        // hole in pyramid center
        printf("  ");

        //create rigth side of pyramid
        printBlocks(numberOfBlocks);

        printf("\n");
    }


    return 0;

}

// function to get valid Height value for mario game
int get_valid_integer()
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);
    return height;
}

// function for printing blocks
void printBlocks(int numberOfBlocks)
{
    for (int i = 0; i < numberOfBlocks; i++)
    {
        printf("#");
    }
}